### PR TITLE
RFC for granting priviledges in snowflake

### PR DESCRIPTION
**Pull request by ryanking**
I am trying to figure out the resource schemas for granting privileges in snowflake.

Below is a doc where I have some examples of how it *could* work. Please give feedback.

### CODE REVIEW

File: examples/priviledges/main.tf [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+
+  schema {
+    schema_name = "bar"
+
+    # these two ^ v are mutually exclusive
+    # database_name = "bar"
+    #  if ^ then it means "all schemas in db"
+
+    # not sure if we should make these mutually exclusive or allow you to create one resource w/ a bunch of privs
+    table {
```
Comment (edulop91) [inline examples/priviledges/main.tf]: not sure if you can have nested blocks like these in the current version of tf

Comment (ryanking) [reply examples/priviledges/main.tf]: i see an example here https://www.terraform.io/docs/providers/aws/r/cloudfront_distribution.html

Comment (edulop91) [reply examples/priviledges/main.tf]: nvm, you can

File: examples/priviledges/main.tf [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+#     USAGE
+# -- For FILE FORMAT, UDF, or SEQUENCE
+#     USAGE
+
+# Given that schema, here are some terraform resources we could support. We should map this out semi-fully, even though
+# we will build incrementally.
+
+# set global privs
+resource "snowflake_priviledge_grants" "a" {
+  name      = "foo1"
```
Comment (edulop91) [inline examples/priviledges/main.tf]: does this name do anything other than provide an ID to track tf state?

Comment (ryanking) [reply examples/priviledges/main.tf]: Just state, because there is no natural ID otherwise.

**Comment by austinylin** [comment]
A few thoughts:

- Thanks for starting this!
- s/priviledges/privileges/g
- Just wanted to check my understanding, this is just how we assign privileges to roles? Users -> roles is another exercise?
- I think generally on the questions about how many permissions to enable to be nested in a single grant we should drive towards simplicity which to me would say let's make it easy to grant a ton of access in a single resource. I think that generally means we end up with one resource per role which is cognitively nice.

**Pull request closed without merging**